### PR TITLE
Adding makefile and pr-hooks config file

### PR DESCRIPTION
**Pull request by hanumantharaomvl**
This PR contains the following changes.
1. Makefile for the Code-Linter job.
2. Pre-commit hooks configuration file for the Code-Linter job.
3. A shell script for the developer to trigger the pre-commit hook validation.

Task Link: https://trello.com/c/H11jBNFR
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Developer Checklist
- [ ] My change requires a change to the documentation.
- [x] I have updated my branch with the latest changes in master.
- [ ] I have communicated with the team any changes that might affect work in progress. 
- [x] I have updated Trello to reflect current state of the feature. 

## Documentation Checklist
- [x] I have updated the necessary README's
- [ ] I have updated the Runbook

## Description of Changes
1. Makefile for the Code-Linter job.
2. Pre-commit hooks configuration file for the Code-Linter job.
3. A shell script for the developer to trigger the pre-commit hook validation.

## Description of Testing Process
Validated the changes aginst multiple repos.

### CODE REVIEW

**Review by philipakash — APPROVED** [review]

File: .pre-commit-pr-hooks.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+repos:
+-   repo: https://github.com/Flux7Labs/central-repo-git-hooks.git
+    rev: feat/US3719
```
Comment (KritikaA06) [inline .pre-commit-pr-hooks.yaml]: Should it be feature here ? or master ?

Comment (hanumantharaomvl) [reply .pre-commit-pr-hooks.yaml]: We kept that branch during testing. Updated to master. @KritikaA06

File: .pre-commit-hooks.yaml [hunk 2]
```diff
@@ -35,15 +35,7 @@
 - id: flake8-lint
   name: flake8-lint
   description: '`flake8` is a command-line utility for enforcing style consistency across Python projects.'
-  entry: flake8  
+  entry: 'flake8 --ignore=E501'
```
Comment (KritikaA06) [inline .pre-commit-hooks.yaml:38]: Why do we want to completely ignore E501? Cannot we define a specific number of characters within a line?

Comment (hanumantharaomvl) [reply .pre-commit-hooks.yaml:38]: what number is in your mind ? Right now we can go with the max no.of chars we observed to fix, but that keep changes.

**Review by max6746 — APPROVED** [review]
LGTM

File: .pre-commit-hooks.yaml (hunk 2 above)
Comment (KritikaA06) [reply .pre-commit-hooks.yaml:38]: > what number is in your mind ? Right now we can go with the max no.of chars we observed to fix, but that keep changes.

I would prefer 120 characters max.

**Pull request merged**